src/main/java/org/ethereum/beacon/discovery/message/FindNodeMessage.java

```diff
@@ -33,18 +36,22 @@ public FindNodeMessage(Bytes requestId, List<Integer> distances) {
     this.distances = distances;
   }
 
-  public static FindNodeMessage fromRlp(List<RlpType> rlpList) {
-    Bytes requestId = Bytes.wrap(((RlpString) rlpList.get(0)).getBytes());
-    RlpList distanceList = (RlpList) rlpList.get(1);
+  private static FindNodeMessage fromRlp(List<RlpType> rlpList) throws DecodeException {
+    if (rlpList.size() != 2) {
+      throw new RlpDecodeException("Invalid RLP list size for FindNode message-data: " + rlpList);
+    }
+    Bytes requestId = RlpUtil.asString(rlpList.get(0), RlpUtil.maxSize(MAX_REQUEST_ID_SIZE));
+    List<RlpType> rlpDistances = RlpUtil.asList(rlpList.get(1));
```

Comment:
Spec doesn't settle any limits here. Though I don't think this is critical, since the total message length is limited and the distance `> 255` is degenerate but still legit. No negative values could be decoded here.
Though your comment gave me an idea to create a test for `FindNodeHandler`